src/databricks/sql/cloudfetch/downloader.py

```diff
@@ -171,3 +202,40 @@ def decompress_data(compressed_data: bytes) -> bytes:
                 uncompressed_data += data
                 start += num_bytes
         return uncompressed_data
+
+
+def http_get_with_retry(url, max_retries=5, backoff_factor=2, download_timeout=60):
```

Comment:
why are we implementing retry behavior here rather than using a Retry passed to the session?

Reply:
agreed. It is in the TODO to also have connection pools